targets/TARGET_NXP/TARGET_MCUXpresso_MCUS/TARGET_LPC546XX/TARGET_FF_LPC546XX/device.h

```diff
@@ -25,8 +25,9 @@
      SYSCON_PDRUNCFG_PDEN_SRAM0_MASK | SYSCON_PDRUNCFG_PDEN_SRAM1_2_3_MASK)
 
 /* Defines used by the sleep code */
-#define LPC_CLOCK_INTERNAL_IRC BOARD_BootClockFRO12M
-#define LPC_CLOCK_RUN          BOARD_BootClockFROHF48M
+#define LPC_CLOCK_INTERNAL_IRC BOARD_BootClockFRO12M()
+#define LPC_CLOCK_RUN          ((SYSCON->DEVICE_ID0 == 0xFFF54628) ? \
```

Comment:
This feels very strange to do. What's the reason that a different clock sequence/function is required for a single part?

Reply:
This is to identify a LPC54628 part in the LPC546XX family. LPC54628 which is capable of running at higher speeds.